rust/src/adapter/datadog_formatter/mod.rs

```diff
@@ -0,0 +1,95 @@
+use std::{time::SystemTime, collections::HashMap};
+
+use rand::Rng;
+use serde::Serialize;
+
+use super::datadog::DatadogConfig;
+
+pub struct DatadogFormatter {
+    pub traces: Vec<Trace>
+}
+
+pub type Trace = Vec<Span>;
+
+#[derive(Default, Debug, Clone, PartialEq, Serialize)]
+pub struct Span {
+    pub trace_id: u64,
+    pub span_id: u64,
+    pub parent_id: u64,
+    pub name: String,
+    pub start: u64,
+    pub duration: u64,
+    pub resource: String,
+    pub error: u8,
+    pub meta: HashMap<String, String>,
+    pub metrics: HashMap<String, f64>,
+    pub service: String,
+    #[serde(rename = "type")]
+    pub typ: String,
+}
+
+impl DatadogFormatter {
+    pub fn new() -> DatadogFormatter {
+        DatadogFormatter {
+            traces: Vec::new(),
+        }
+    }
+
+    pub fn add_trace(&mut self, trace: Trace) {
+        self.traces.push(trace)
+    }
+}
+
+impl Span {
+    pub fn new(
+        config: DatadogConfig,
+        trace_id: u64,
+        parent_id: Option<u64>,
+        name: String,
+        start_time: SystemTime,
+        end_time: SystemTime,
+    ) -> Span {
+        let span_id = new_span_id();
+
+        let p_id = match parent_id {
+            Some(id) => id,
+            None => new_span_id(),
+        };
+
+        let mut meta = config.default_tags.clone();
+
+        meta.insert("http.status_code".to_string(), "200".to_string());
+        meta.insert("http.method".to_string(), "POST".to_string());
+        meta.insert("http.url".to_string(), "http://localhost:3000".to_string());
+
+        Span {
+            trace_id,
+            span_id,
+            parent_id: p_id,
+            name,
+            meta,
+            metrics: HashMap::new(),
+            service: "planktonic".to_string(),
```

Comment:
TODO: switch this with service_name from config and figure out other ways to pass down metadata.